src/Fantomas.Tests/UtilsTests.fs

```diff
@@ -112,3 +113,65 @@ SetupTesting.generateSetupScript __SOURCE_DIRECTORY__
 #endif
 """
     |> mergeAndCompare a b
+
+[<Test>]
+let ``when input is empty`` () =
+    let property (p: bool) : bool =
+        let before, after = List.partitionWhile (fun _ _ -> p) []
+        before = [] && after = []
+
+    Check.QuickThrowOnFailure(property true)
+    Check.QuickThrowOnFailure(property false)
+
+[<Test>]
+let ``when predicate always returns false`` () =
+    let property (xs: int list) : bool =
+        let before, after =
+            List.partitionWhile (fun _ _ -> false) xs
+
+        before = [] && after = xs
+
+    Check.QuickThrowOnFailure property
+
+[<Test>]
+let ``when predicate always returns true`` () =
+    let property (xs: int list) : bool =
+        let before, after =
+            List.partitionWhile (fun _ _ -> true) xs
+
+        before = xs && after = []
+
+    Check.QuickThrowOnFailure property
+
+[<Test>]
+let ``when predicate returns true until certain index`` () =
+    let property (xs: int list, i: int) : bool =
+        let before, after =
+            List.partitionWhile (fun index _ -> i <> index) xs
+
+        let beforeLength = List.length before
+        let afterLength = List.length after
+
+        beforeLength = i
+        && afterLength = List.length xs - i
+        && before @ after = xs
+
+    let gen =
+        gen {
+            let! xs =
```

Comment:
Tiny point, you don't have to change anything but I do think it's useful for learning the right mental model: it's generally better to *construct* the right cases rather than to *filter down* to them. Here, you could generate `x : int` and `xs : int list`, and then return `x :: xs`, constructing a nonempty list.

Reply:
The reason it's better to construct rather than filter is because:

* it's easier to reason about the distribution of the resulting test cases
* you make FsCheck do less wasted computation in creating test cases only to discard them immediately.

In this case, neither of those is a big deal, but if your condition is hard to satisfy or you're specifically testing some edge case where you want precise control over the distribution, it's better to be in the habit of constructing cases which automatically have the right property.

Reply:
It turns out that `Gen.nonEmptyListOf` is a thing, by the way.

Reply:
This is great - thank you! I actually already hit an issue yesterday where I attempted to filter in a way that caused FsCheck to get "stuck" trying to find a compliant value.